src/RobotInterface/YarpImplementation/include/BipedalLocomotion/RobotInterface/YarpSensorBridgeImpl.h

```diff
@@ -977,51 +977,77 @@ struct YarpSensorBridge::Impl
             {
-                ok = ok
-                     && devList[devIdx]->poly->view(controlBoardRemapperInterfaces.motorEncoders);
-                ok = ok && devList[devIdx]->poly->view(controlBoardRemapperInterfaces.currsensors);
+                okMotorsSensor
+                    = devList[devIdx]->poly->view(controlBoardRemapperInterfaces.motorEncoders);
+                okMotorsSensor
+                    = okMotorsSensor
+                      && devList[devIdx]->poly->view(controlBoardRemapperInterfaces.currsensors);
             }
-            if (metaData.bridgeOptions.isPIDsEnabled)
+            if (!okPID && metaData.bridgeOptions.isPIDsEnabled)
             {
-                ok = ok && devList[devIdx]->poly->view(controlBoardRemapperInterfaces.pids);
+                okPID = devList[devIdx]->poly->view(controlBoardRemapperInterfaces.pids);
             }
 
-            if (ok)
+            if (okPID && okJointsSensor && okMotorsSensor && okPWM)
             {
-                break;
-            }
-        }
+                if (!compareControlBoardJointsList())
+                {
+                    log()->error("{} Could not attach to remapped control board interface.",
+                                 logPrefix);
+                    return false;
+                }
 
-        if (!ok)
-        {
-            log()->error("{} Could not find a remapped remote control board with the desired "
-                         "interfaces",
-                         logPrefix);
-            return false;
+                return true;
+            }
         }
 
-        if (!compareControlBoardJointsList())
-        {
-            log()->error("{} Could not attach to remapped control board interface.", logPrefix);
-            return false;
-        }
+        auto isSensorRequired = [](bool isRequired) -> const char* {
+            return isRequired ? "required" : "not required";
+        };
+
+        auto isSensorFound
+            = [](bool isFound) -> const char* { return isFound ? "found" : "not found"; };
+
+        log()->error("{} Could not find a remapped remote control board with the desired "
+                     "interfaces. Here the status of the interfaces. "
+                     "Joint sensors: {} - {}, "
+                     "Motor sensors: {} - {}, ",
+                     "PID sensors: {} - {}, ",
+                     "PWM sensors: {} - {}.",
+                     logPrefix,
+                     isSensorFound(okJointsSensor),
+                     isSensorRequired(metaData.bridgeOptions.isJointSensorsEnabled),
+                     isSensorFound(okMotorsSensor),
+                     isSensorRequired(metaData.bridgeOptions.isMotorSensorsEnabled),
+                     isSensorFound(okPID),
+                     isSensorRequired(metaData.bridgeOptions.isPIDsEnabled),
+                     isSensorFound(okPWM),
+                     isSensorRequired(metaData.bridgeOptions.isPWMControlEnabled));
```

Comment:
I suspect that this error message could be a bit misleading. Suppose that ``metaData.bridgeOptions.isJointSensorsEnabled = false``, we will have
```
Joint sensors: found - not required,
```
but actually we never checked if the joint sensor interfaces are actually found. In my opinion, it would be better to have only
```
Joint sensors: not required,
```

Reply:
What do you think of https://github.com/ami-iit/bipedal-locomotion-framework/pull/608/commits/c8c6e0c23a5fc99c5a5112d300a069bbcb69cf79 ?

Reply:
Looks good!